### ISSUE TITLE
trial.suggest_int is not sticking to step size

### ISSUE THREAD

**Issue by Username159** [optuna/optuna#2467]
## Expected behavior
I am using Optuna to optimize the hyperparameters for my CNNLSTM neural net. 
I am using Pytorch.
I am using suggest_int to suggest a number for the kernels of my two Conv2-layers and the max pooling layer.

```
kernel1 = trial.suggest_int('kernel1',1,7, step=2)
kernel2 = trial.suggest_int('kernel2',1,7, step=2)
kernel_pool = trial.suggest_int('kernel_pool',3,5, step=2)
```

I am using a TPESampler
`sampler = optuna.samplers.TPESampler(seed=10, n_startup_trials = 15, multivariate=True)`
and a PercentilePruner
`pruner = optuna.pruners.PercentilePruner(65.0, n_startup_trials=10, n_warmup_steps=175, interval_steps=3)`

And I optimize my objective function like this:
`study.optimize(objective_week, n_trials=15,  gc_after_trial=True, catch=(RuntimeError,))`.

I have three other hyperparamters where I use trial.suggest_int
```
layers = trial.suggest_int('layers',2,4)
hidden_states = trial.suggest_int('hidden_states',64,256, step=32)
epochs = trial.suggest_int('epochs', 600, 1400, step=200)
```
and got no problems with them yet.

**Since trial 22 I am getting 2,4,6 (resp. 4 for kernel_pool) as suggested values for the kernels1/2 and no longer 1,3,5,7 (resp. 3,5 for kernel_pool).**

For the other hyperparameter I use suggest_float, suggest_categorical, no problems there.

## Environment

- Optuna version: 2.5.0
- Python version: 3.9.1
- OS: Linux Manjaro
- (Optional) Other libraries and their versions:
torch: 1.7.1

## Steps to reproduce

1.
2.
3.

## Reproducible examples (optional)

```python
# python code
```

**Comment by nzw0301** [comment]
@Username159 Please update your optuna version to the latest stable version: `2.6.0`. I think the same issue was reported in https://github.com/optuna/optuna/issues/2391 and optuna has fixed it recently.

**Comment by Username159** [comment]
Updated to the `2.6.0` version yesterday, but still facing the same problem today.

**Comment by nzw0301** [comment]
Could you share the code?

**Comment by Username159** [comment]
```

  # Definitions
  def define_model_week(trial):
      filter1 = trial.suggest_int('filter1',80,320, step=16)
      filter2 = trial.suggest_int('filter2',filter1//4*1, filter1//4*3, step = filter1//4*1)

      layers = trial.suggest_int('layers',2,4)

      cnn_drop = 0.05
      before_lin_drop = 0.01
      lstm_drop = 0.05
      emb_drop = 0.05
    
      hidden_states = trial.suggest_int('hidden_states',64,256, step=32)

      kernel1 = trial.suggest_int('kernel1',1,7, step=2)
      kernel2 = trial.suggest_int('kernel2',1,7, step=2)
      kernel_pool = trial.suggest_int('kernel_pool',3,5, step=2)

      model = CNNLSTM(
                in_channel = v_dict['in_channel'],
                filter1 = filter1, #v_dict['filter1'],
                kernel1 = (kernel1, kernel1),
                filter2 = filter2, #v_dict['filter2'],
                kernel2 = (kernel2, kernel2),
                kernel_pool = (kernel_pool, kernel_pool),
                cnn_drop = cnn_drop,
                before_lin_drop = before_lin_drop,
                T = v_dict['T'],                
                hidden_states = hidden_states,
                layers = layers,
                total_features = v_dict['total_features'],
                lstm_drop = lstm_drop,
                output_size = v_dict['output_size'],
                len_forecast = v_dict['len_forecast'],
                emb_dims = v_dict['emb_dims'],
                emb_dropout = emb_drop,
                cat_names = v_dict['cat_names'],
                leaky=v_dict['leaky']
                )
      
      return model

# similar model definition for CNNBLSTM
 
  def objective_week(trial):
    global max_lrs
    global train_accs_list
    global swa_accs_list

    # Generate the model.
    model_type = trial.suggest_categorical('model_type', ['CNNLSTM', 'CNNBLSTM'])

    if model_type == 'CNNLSTM':
      torch.manual_seed(10)
      model = define_model_week(trial).to(device)
      batch_size = 256
    
    elif model_type == 'CNNBLSTM':
      torch.manual_seed(10)
      model = define_model_cnnblstm_week(trial).to(device)
      batch_size = 128


    criterion = nn.SmoothL1Loss(reduction='sum', beta=3.0)

    # Get the dataloaders.

    dls_train = DataLoader(small_dataset_train, batch_size=batch_size, shuffle=False, drop_last=False, num_workers=0)
    dls_test = DataLoader(small_dataset_test, batch_size=batch_size, shuffle=False, drop_last=False, num_workers=0)

    # Generate the optimizers.
    # optimizer_name = trial.suggest_categorical('optimizer_name', ['SGD', 'DiffGrad'])
    optimizer_name = trial.suggest_categorical('optimizer_name', ['DiffGrad'])
    weight_decay = 0.05

    if optimizer_name =='SGD':
       # see 'DiffGrad'

    if optimizer_name =='DiffGrad':
      scheduler_name = trial.suggest_categorical('scheduler_name', ['OneCycleLR', 'LambdaLR', 'ReduceOnPlateau'])
      if scheduler_name == 'OneCycleLR':
       # similar to 'ReduceOnPlateau'
      elif scheduler_name == 'LambdaLR':
       # similar to 'ReduceOnPlateau'
      elif scheduler_name == 'ReduceOnPlateau':
        epochs = trial.suggest_int('epochs', 1000, 2000, step=200)
        momentum = trial.suggest_float('momentum', 0.7, 0.99)
        base_momentum = trial.suggest_float('base_momentum', 0.85,0.85)
        optimizer = getattr(torch_optim, 'DiffGrad')(model.parameters(), lr=1, weight_decay=weight_decay)

        lr_finder = LRFinder(model, optimizer, criterion, device=device)
        print()
        lr_finder.range_test(dls_train, dls_test, start_lr = 1e-7, end_lr=1, num_iter=200, smooth_f=0.05)
        max_lr = lr_finder.history['lr'][np.gradient(np.array(lr_finder.history['loss'])).argmin()]
        max_lrs.append(max_lr)
        lr_finder.reset()
        del lr_finder
        
        print(f'After learning rate finder. max_lr = {max_lr}')
        print()
        
        del optimizer
        step_no = trial.suggest_int('step_no', 3,3,step=2)
        optimizer = getattr(torch_optim, 'DiffGrad')(model.parameters(), lr=max_lr, weight_decay=weight_decay)
        scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, mode='max', factor=0.5, patience=int(epochs*0.15)*len(dls_train), threshold=0.01, threshold_mode='rel', cooldown=0, min_lr=1e-10, eps=1e-08, verbose=False)



    # Training of the model.
    swa_model = torch.optim.swa_utils.AveragedModel(model)
    swa_start = int(epochs*0.85) #*len(dls_train)
    set_swa_scheduler = True
    swa_scheduler = None

    acc_train_general, acc_test_general = 0.0, 0.0
    
    for epoch in tqdm(range(epochs)):
      model.train()
      for X_cont_cat_train, y_train  in dls_train:
      # training loop
      # .....


      # Validation of the model.
      # ....

      acc_train_general, acc_test_general = accuracy_func_dls_general_mod(model, dls_train, dls_test, p=0.2)
      .....
      trial.report(accuracy, epoch)

      # Handle pruning based on the intermediate value.
      if trial.should_prune():
       ......


    return accuracy


# Create study
sampler = optuna.samplers.TPESampler(seed=10, n_startup_trials = 17, multivariate=True)  
pruner = optuna.pruners.PercentilePruner(65.0, n_startup_trials=10, n_warmup_steps=175, interval_steps=3)

study = optuna.create_study(direction="maximize", study_name=new_study_name, load_if_exists=False, sampler = sampler, pruner=pruner)


# Run study
study.optimize(objective_sanity, n_trials=35, , gc_after_trial=True, catch=(RuntimeError,))



```

That's a simplified and abbreviated version of my code.

And then after some trials it started that i got something like this for example:

number | value | max_lr | params_base_momentum | params_epochs | params_filter1 | params_filter2 | params_hidden_states | params_kernel1 | params_kernel2 | params_kernel_pool | params_layers | params_model_type | params_momentum | params_optimizer_name | params_scheduler_name | params_step_no | state | duration
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- 
27 | 90.67407 | 0.00517 | 0.85 | 1800 | 160 | 40 | 224 | 2 | 6 | 4 | 4 | CNNLSTM | 0.7427 | DiffGrad | ReduceOnPlateau | 3 | COMPLETE | 1 h  1 min


A 2 for kernel1, a 6 for kernel2 and a 4 for kernel_pool. And that shouldn't be possible.

**Comment by nzw0301** [comment]
I'm afraid that we cannot investigate the code including some custom class such as `CNNLSTM` and undefined code block such as training `for-loop`. Could you share reproducible code?
